.github/workflows/reusable_build.yml

```diff
@@ -0,0 +1,75 @@
+### For the pure soul wishes to move it to another place
+# https://github.com/orgs/community/discussions/9050
+
+name: Build ClickHouse
+'on':
+  workflow_call:
+    inputs:
+      build_name:
+        description: the value of build type from tests/ci/ci_config.py
+        required: true
+        type: string
+      checkout_depth:
+        description: the value of the git shallow checkout
+        required: false
+        type: number
+        default: 1
+      runner_type:
```

Comment:
to stay on a safe side and have this workflow more generic I'd suggest avoid using default runner_type value and pass it explicitly whenever wf is called.

Reply:
not sure why. We use only `builder` here, and the default is safe even for other places